models/articles.model.js

```diff
@@ -23,12 +45,46 @@ exports.selectAllArticles = () => {
       articles.article_img_url,
       COUNT(comments.article_id)::INT AS comment_count
     FROM articles
-    LEFT JOIN comments ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;
-  `;
+    LEFT JOIN comments ON articles.article_id = comments.article_id`;
 
-  return db.query(query).then(({ rows }) => {
+  if (author) {
+    queryValues.push(author);
+    queryString += ` WHERE articles.author = $${queryValues.length}`;
+  }
+  if (topic) {
+    queryValues.push(topic);
+    if (queryValues.length > 1) {
+      queryString += ` AND articles.topic = $${queryValues.length}`;
+    } else {
+      queryString += ` WHERE articles.topic = $${queryValues.length}`;
+    }
+  }
+  queryString += ` 
+  GROUP BY articles.article_id 
+  ORDER BY articles.created_at DESC;`;
+
+  return db.query(queryString, queryValues).then(({ rows }) => {
     return rows;
   });
 };
+
+exports.selectArticleComments = (article_id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1;`, [article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not found" });
+      }
+
+      return db.query(
+        `SELECT comment_id, votes, created_at, author,body, article_id
+        FROM comments
+        WHERE article_id = $1
+        ORDER BY created_at DESC;`,
+        [article_id]
+      );
+    })
```

Comment:
This is a fine way of going about it, but you could refactor to extract out the select articles as it's own model and use a Promise.all in the controller to call both models at the same time. Check the advanced error notes!